Controller/src/LSST/M1M3/SS/Accelerometer/AccelerometerSettings.cpp

```diff
@@ -8,15 +8,19 @@
 #include <AccelerometerSettings.h>
 #include <pugixml.hpp>
 #include <boost/lexical_cast.hpp>
+#include <Log.h>
 
 namespace LSST {
 namespace M1M3 {
 namespace SS {
 
 void AccelerometerSettings::load(const std::string &filename) {
 	pugi::xml_document doc;
-	doc.load_file(filename.c_str());
-
+	pugi::xml_parse_result load_file_xml_parse_result = doc.load_file(filename.c_str());
```

Comment:
Consider using `auto` here, as in: `auto load_file_xml_parse_result =...`
as the type is not useful information for the reader.

Reply:
You had to write this same 6 lines nearly 20 times. I *strongly* encourage you to write a single function that performs this operation, then call it in all these places.

Reply:
I agree with you. The change was very easy to do by using find and replace in files, but having a single function makes the maintenance easier. I moved this to a static method to SettingReader class, which sounded to me like the most general one for that aspect.

Reply:
Maybe, all classes reading settings should inherit from a base class with this general method, but I am not sure if such a change is required now. Anyway, this was the purpose of putting my hands on, to learn how the code is designed and written.

Reply:
A static member function or free function seems fine for something so simple.

If the classes all had a log member variable (and perhaps they should, but that seems out of scope for this change) then I think it would make sense to use inheritance and define the function in a base class.